### PR TITLE
Fixes Carrier being able to drop 10 huggers in a group of marines in about a second

### PR DESCRIPTION
**Pull request by terranaut1**
## About The Pull Request

3 second cooldown for carriers to pull huggers out of their storage

## Why It's Good For The Game

you know if you've seen it

## Changelog
:cl:Terranaut
balance: 3 Second Cooldown for Carriers taking a Hugger out of storage
/:cl:

### CODE REVIEW

**Comment by jroinc** [comment]
Agree in principle, but I think 3 seconds is too much and will cause issues in combat. I'd rather see a 2 second delay, but let's see the TM first.

**Comment by GravitronBlue** [comment]
Should double check if this PR works, was testmerged earlier and I didn't notice any sort of cooldown as a carrier.

**Comment by terranaut1** [comment]
oops haha actually works now :)

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 1]
```diff
@@ -7,12 +7,9 @@
...
 	ability_name = "throw facehugger"
 	keybind_signal = COMSIG_XENOABILITY_THROW_HUGGER
+	cooldown_timer = 3 SECONDS
 
-/datum/action/xeno_action/activable/throw_hugger/get_cooldown()
-	var/mob/living/carbon/xenomorph/carrier/X = owner
-	return X.xeno_caste.hugger_delay
-
-/datum/action/xeno_action/activable/throw_hugger/can_use_ability(atom/A, silent = FALSE, override_flags = XACT_IGNORE_COOLDOWN) // true
+/datum/action/xeno_action/activable/throw_hugger/can_use_ability(atom/A, silent = FALSE) // true
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm]:
```suggestion
/datum/action/xeno_action/activable/throw_hugger/can_use_ability(atom/A, silent = FALSE, override_flags) // true
```

Comment (terranaut1) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm]: done

**Comment by terranaut1** [comment]
there's an argument to be made for some further adjustments, setting the cooldown to 2 or seperating the pull hugger out of storage and throw hugger actions, or removing the cooldown from throwing one, since currently the whole cycle from
>start
>take hugger
>throw hugger
>start
is 6 seconds which seems a bit long

**Comment by Rohesie** [comment]
Up to you. I'll leave this open until you are satisfied with it.

**Comment by terranaut1** [comment]
Tm it and I'll ask for feedback in a couple days if you will

**Comment by BurgerLUA** [comment]
This is unironically one of the worst PRs that I've ever come across. It makes carrier absolutely unplayable in situations other than "The first 10 seconds of LZ1 camping." which should've been nerfed instead.

**Comment by terranaut1** [comment]
You're welcome to provide ideas you think are better or expand on one of the suggestions I already made myself but if you think a carrier running through a group of Marines and mashing middle mouse and q as fast as he is able to to infect literally all of them you're a fool.

**Comment by BurgerLUA** [comment]
How to fix carrier:
1. Make traps not have visual glitches/exploits.

That is how you fix carrier. You don't make it more clunky to play by already nerfing a ultra-nerfed ability.

**Comment by terranaut1** [comment]
This PR isn't about traps but being able to 6v1 Marines in a few seconds and infecting them all. Go away if you just want to salt post.

**Comment by BurgerLUA** [comment]
you asked me, personally, on how to fix carrier. give you an answer. you tell me to fuck off and claim im being salty

the absolute state of developement

**Comment by Rohesie** [comment]
One possibility here is making each action take a short time. Even `0.5` seconds delay makes a difference, you see.

**Comment by terranaut1** [comment]
That'd be fine for the pulling a hugger out but if we did that on the throw it'd completely neuter the ability I feel like. Maybe that's just me though, I really hate having movement of my character impaired for any reason.

**Comment by Rohesie** [comment]
Up to you. Was just spitballing suggestions. Maybe gather more opinions first.

**Comment by terranaut1** [comment]
Uses castedatum hugger_delay now (changing the cooldown to 2.5/2/1.5/1). Only taking a hugger out of storage triggers the cooldown now, throwing it does not.

**Comment by Rohesie** [comment]
Let us know when you feel this PR has been tested enough and is ready for merge.

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 2]
```diff
@@ -7,12 +7,13 @@
 	mechanics_text = "Click once to bring a facehugger into your hand. Click again to ready that facehugger for throwing at a target or tile."
 	ability_name = "throw facehugger"
 	keybind_signal = COMSIG_XENOABILITY_THROW_HUGGER
+	cooldown_timer = 3 SECONDS
 
-/datum/action/xeno_action/activable/throw_hugger/get_cooldown()
-	var/mob/living/carbon/xenomorph/carrier/X = owner
+/datum/action/xeno_action/activable/throw_hugger/get_cooldown()
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm]:
```suggestion
/datum/action/xeno_action/activable/throw_hugger/get_cooldown()
```

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 3]
```diff
@@ -7,12 +7,13 @@
 	mechanics_text = "Click once to bring a facehugger into your hand. Click again to ready that facehugger for throwing at a target or tile."
 	ability_name = "throw facehugger"
 	keybind_signal = COMSIG_XENOABILITY_THROW_HUGGER
+	cooldown_timer = 3 SECONDS
 
-/datum/action/xeno_action/activable/throw_hugger/get_cooldown()
-	var/mob/living/carbon/xenomorph/carrier/X = owner
+/datum/action/xeno_action/activable/throw_hugger/get_cooldown()	
+	var/mob/living/carbon/xenomorph/carrier/X = owner
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm]:
```suggestion
	var/mob/living/carbon/xenomorph/carrier/X = owner
```

**Pull request merged**